src/test/java/seedu/address/model/person/FriendNameTest.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class FriendNameTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new FriendName(null));
+    }
+
+    @Test
+    public void constructor_invalidName_throwsIllegalArgumentException() {
+        String invalidName = "";
+        assertThrows(IllegalArgumentException.class, () -> new FriendName(invalidName));
+    }
+
+    @Test
+    public void isValidName() {
+        // null name
+        assertThrows(NullPointerException.class, () -> FriendName.isValidFriendName(null));
+
+        // invalid name
+        assertFalse(FriendName.isValidFriendName("")); // empty string
+        assertFalse(FriendName.isValidFriendName(" ")); // spaces only
+        assertFalse(FriendName.isValidFriendName("^")); // only non-alphanumeric characters
+        assertFalse(FriendName.isValidFriendName("peter*")); // contains non-alphanumeric characters
+
+        // valid name
+        assertTrue(FriendName.isValidFriendName("peter jack")); // alphabets only
+        assertTrue(FriendName.isValidFriendName("12345")); // numbers only
```

Comment:
Is this intended behaviour? Can a friend name be purely numeric?

Reply:
I agree, I think changing it such that at least the first character is a letter makes sense as there are people with the name "X Æ A-12" (Elon musk's child).